src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/ReactiveCosmosQueryExecution.java

```diff
@@ -39,6 +43,31 @@ public Object execute(DocumentQuery query, Class<?> type, String container) {
         }
     }
 
+    final class SingleEntityExecution implements ReactiveCosmosQueryExecution {
+
+        private final ReactiveCosmosOperations operations;
+        private final ReturnedType returnedType;
+
+        public SingleEntityExecution(ReactiveCosmosOperations operations, ReturnedType returnedType) {
+            this.operations = operations;
+            this.returnedType = returnedType;
+        }
+
+        @Override
+        public Object execute(DocumentQuery query, Class<?> type, String container) {
+            return operations.find(query, type, container)
+                .buffer(2)
+                .map((vals) -> {
+                    if (vals.size() > 1) {
+                        throw new CosmosDBAccessException("Too many results - return type "
+                            + returnedType.getReturnedType() +
+                            " is not of type Iterable but find returned multiple results");
```

Comment:
`find` should be `found`

Reply:
changed to a more reactive appropriate error message...
"Too many results - Expected Mono<"
                            + returnedType.getReturnedType() +
                            "> but query returned multiple results"